code/modules/mob/living/simple_animal/hostile/wastebots.dm

```diff
@@ -70,3 +74,84 @@
 	projectiletype = /obj/item/projectile/beam/laser/pistol
 	faction = list("wastebot")
 	check_friendly_fire = 1
+
+/mob/living/simple_animal/hostile/handy/sentrybot
+	name = "sentry bot"
+	desc = "A pre-war military robot armed with a deadly gatling laser and covered in thick armor plating."
+	icon_state = "sentrybot"
+	icon_living = "sentrybot"
+	icon_dead = "sentrybot"
+	health = 360
+	maxHealth = 360
+	healable = 0
+	speed = 1
+	melee_damage_lower = 85
+	melee_damage_upper = 85
+	rapid = 2 //6 projectiles. oof.
+	ranged_cooldown_time = 15 //brrrrrrrrrrrrt
+	ranged = 1
+	retreat_distance = 0
+	minimum_distance = 0
+	attacktext = "pulverizes"
+	attack_sound = 'sound/weapons/punch1.ogg'
+	projectilesound = 'sound/weapons/laser.ogg'
+	projectiletype = /obj/item/projectile/beam/laser/pistol/weak
+	faction = list("wastebot")
+	check_friendly_fire = 1
+
+/obj/item/projectile/beam/laser/pistol/weak
+	damage = 10 //quantity over quality
+
+/mob/living/simple_animal/hostile/handy/sentrybot/bullet_act(obj/item/projectile/Proj)
+	if(!Proj)
+		return
+	if(prob(10))
+		visible_message("<span class='danger'>\The [src] releases a defensive flashbang!</span>")
+		sleep(3)
+		var/flashbang_turf = get_turf(src)
+		if(!flashbang_turf)
+			return
+		for(var/mob/living/M in get_hearers_in_view(4, flashbang_turf))
+			bang(get_turf(M), M)
+	if(prob(75) || Proj.damage > 26) //prob(x) = chance for proj to actually do something, adjust depending on how OP you want sentrybots to be
+		return ..()
+	else
+		visible_message("<span class='danger'>\The [Proj] bounces off \the [src]'s armor plating!</span>")
+		return 0
+
+/mob/living/simple_animal/hostile/handy/sentrybot/proc/bang(turf/T , mob/living/M)
+	if(M.stat == DEAD)	//They're dead!
+		return
+	M.show_message("<span class='warning'>BANG</span>", 2)
+	playsound(loc, 'sound/weapons/flashbang.ogg', 100, 1)
+	var/distance = max(0,get_dist(get_turf(src),T))
+
+//Flash
+	if(M.flash_act(affect_silicon = 1))
+		M.Knockdown(max(200/max(1,distance), 60))
+//Bang
+	if(!distance || loc == M || loc == M.loc)
+		M.Knockdown(200)
+		M.soundbang_act(1, 200, 10, 15)
+
+	else
+		M.soundbang_act(1, max(200/max(1,distance), 60), rand(0, 5))
+
+/mob/living/simple_animal/hostile/handy/sentrybot/Destroy()
```

Comment:
You'll want to handle this before `qdel()` is invoked, else you risk lagging the whole subsystem with those `sleep()` procs.
Ideally this is in a `set waitfor = FALSE` proc with `stoplag()` instead of `sleep()`, handled on `death()`, not `Destroy()`.